src/elasticsearch/elasticsearch.ts

```diff
@@ -1,25 +1,40 @@
 import {Instance} from '../aws/types';
-import {
-    Documents,
-    ElasticsearchClusterStatus,
-    ElasticsearchNode,
-    NodeStats,
-} from './types';
+import {Documents, ElasticsearchClusterStatus, ElasticsearchNode, NodeStats,} from './types';
 import {ssmCommand} from '../utils/ssmCommand';
+import {retry} from '../utils/helperFunctions';
 
-export function getClusterHealth(instanceId: string): Promise<ElasticsearchClusterStatus> {
-    return ssmCommand('curl localhost:9200/_cluster/health', instanceId)
-        .then((result: string) => {
-            const clusterStatus: ElasticsearchClusterStatus = JSON.parse(result);
-            return clusterStatus;
-        })
-}
+export class Elasticsearch {
+    instanceId: string;
+
+    constructor(instanceId: string) {
+        this.instanceId = instanceId;
+    }
 
-export function getElasticsearchNode(instance: Instance): Promise<ElasticsearchNode> {
-    console.log(`Searching for Elasticsearch node with private ip: ${instance.privateIp}`);
-    return ssmCommand(`curl localhost:9200/_nodes/${instance.privateIp}`, instance.id)
-        .then((result: string) => {
-            const json = JSON.parse(result);
+    execute(commandPath: string, requestJson?: object): Promise<any> {
+        const requestJsonSettings = requestJson ? `-X PUT -H 'Content-Type: application/json' -d '${JSON.stringify(requestJson)}'` : "";
+        return ssmCommand(`curl localhost:9200${commandPath} ${requestJsonSettings}`, this.instanceId)
+            .then((result: string) => {
+                return JSON.parse(result);
+            })
+    }
+
+    executeCommand(commandPath: string, requestJson: object, taskName: string): Promise<boolean> {
+        return retry(() => this.execute(commandPath, requestJson).then((jsonResult: any) => {
+            if (!(jsonResult.acknowledged === true)) {
+                throw `unexpected Elasticsearch response when attempting to ${taskName}, we got: ${jsonResult}`;
```

Comment:
Something we missed when we were doing this PR - `jsonResult` is a Javascript [`Object`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object), which doesn't automatically become a helpful string! [The logs](https://logs.gutools.co.uk/s/ophan/goto/06499fa5f7753da98efbd064e1760119) just say `[object Object]`:

> Failed when setting cluster setting cluster.routing.rebalance.enable to 'none' with : unexpected Elasticsearch response when attempting to setting cluster setting cluster.routing.rebalance.enable to 'none', we got: [object Object]

We should have used [`JSON.stringify(jsonResult)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) - I'll add a PR for that!